src/utilstrencodings.cpp

```diff
@@ -528,3 +528,39 @@ std::optional<std::vector<uint8_t>> ParseArbitraryInt(const std::string& num_str
     }
     return result;
 }
+
+std::string ArbitraryIntStr(std::vector<uint8_t> bytes)
+{
+    // Only serialize up to the most significant non-zero byte.
+    size_t end = bytes.size();
+    for (; end > 0  && bytes[end - 1] == 0; --end) {}
+
+    std::string result;
+    while (end > 0) {
+        // Write out the next lowest digit.
+        auto nextDigit = bytes[0] % 10;
+        result = tfm::format("%d%s", nextDigit, result);
```

Comment:
Why is this correct? x mod 10 is not the same as (x mod 256) mod 10. Counterexample: 256.

Reply:
I had the two operations ("write out next digit" and "divide by 10") in the wrong order. The unit tests I thought I'd already implemented but hadn't, caught this issue. Fixed now.